File: CppProject/gfg/pattern.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> pattern(int N){
    // code here
    vector<int> ans;
    int num = N;
    bool isDecreasing = true;
    while(true) {
        if(num > 0 && isDecreasing) {
            ans.push_back(num);
            num = num-5;
        } else {
            isDecreasing = false;
            if(num > N){
                break;
            }
            ans.push_back(num);
            num = num+5;
        }
    }
    return ans;
}

int main() {
    vector<int> result = pattern(16);
    for (int num : result) {
        cout << num << " ";
    }
    cout << endl;
    return 0;
}

File: PythonProject/gfg/pattern.py
def pattern(N):
    # code here
    ans = []
    num = N
    isDecreasing = True
    
    while True:
        if num > 0 and isDecreasing:
            ans.append(num)
            num = num - 5
        else:
            isDecreasing = False
            if num > N:
                break
            ans.append(num)
            num = num + 5
            
    return ans

n = 16
result = pattern(16)
print(result)